### PR TITLE
fix(developer): font file meta data is invalid error

### PR DESCRIPTION
**Pull request by markcsinclair**
Added a new keyboard info compiler error message (ERROR_FontFileMetaDataIsInvalid) to capture ttfMeta.promise errors when trying to extract the font family from a file. With the addition of this error, the ERROR_FontFileCannotBeRead error is unlikely to be triggered, as this would require parsable meta data that nevertheless did not contain the font family. Accordingly, it has been marked as covered in the code.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-keyboard-info/src/font-family.ts [hunk 1]
```diff
@@ -16,12 +16,7 @@ export async function getFontFamily(source: Uint8Array) {
   }
 
   const buffer = Buffer.from(source);
-  let font = null;
-  try {
-    font = await ttfMeta.promise(buffer);
-  } catch(e) {
-    return null;
-  }
+  const font = await ttfMeta.promise(buffer);
```
Comment (mcdurdin) [inline developer/src/kmc-keyboard-info/src/font-family.ts:19]: We need to update the comment on line 9 because an invalid font may now throw an exception.

**Review by mcdurdin — APPROVED** [review]
LGTM. I am assuming you checked all callers to `getFontFamily` to ensure that exceptions are handled?

File: developer/src/kmc-keyboard-info/src/font-family.ts (hunk 1 above)
Comment (markcsinclair) [reply developer/src/kmc-keyboard-info/src/font-family.ts:19]: getFontFamily() is only called in one place, from within KeyboardInfoCompiler.fontSourceToKeyboardInfoFont(), where we now catch the excptions (that could always have been thrown).  I have added a @throws comment.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.292-beta](https://keyman.com/downloads/releases/beta/17.0.292)

**Pull request merged**